tests/test_tasks.py

```diff
@@ -1,118 +1,169 @@
-import asyncio
-import logging
-import time
-from asyncio import AbstractEventLoop
-from typing import TYPE_CHECKING, Any, Dict, List, NoReturn, Tuple
+from typing import Callable, NoReturn
+from unittest.mock import AsyncMock, call, patch
 
 import pytest
-from pytest import LogCaptureFixture
-
-if TYPE_CHECKING:
-    from pytest.capture import CaptureFixture
-else:
-    CaptureFixture = Any
 
 from fastapi_restful.tasks import repeat_every
 
-logging.basicConfig(level=logging.INFO)
-
-
-def ignore_exception(_loop: AbstractEventLoop, _context: Dict[str, Any]) -> None:
-    pass
-
-
-@pytest.fixture(autouse=True)
-def setup_event_loop(event_loop: AbstractEventLoop) -> None:
-    event_loop.set_exception_handler(ignore_exception)
-
-
-@pytest.mark.asyncio
-async def test_repeat_print(capsys: CaptureFixture) -> None:
-    @repeat_every(seconds=0.01, max_repetitions=3)
-    async def repeatedly_print_hello() -> None:
-        print("hello")
-
-    await repeatedly_print_hello()
-    await asyncio.sleep(0.1)
-    out, err = capsys.readouterr()
-    assert out == "hello\n" * 3
-    assert err == ""
+# Fixtures:
 
 
-@pytest.mark.asyncio
-async def test_repeat_print_delay(capsys: CaptureFixture) -> None:
-    @repeat_every(seconds=0.07, max_repetitions=3)
-    def repeatedly_print_hello() -> None:
-        print("hello")
+@pytest.fixture(scope="module")
+def seconds() -> float:
+    return 0.01
 
-    await repeatedly_print_hello()
-    await asyncio.sleep(0.1)
-    out, err = capsys.readouterr()
-    assert out == "hello\n" * 2
-    assert err == ""
 
+@pytest.fixture(scope="module")
+def max_repetitions() -> int:
+    return 3
 
-@pytest.mark.asyncio
-async def test_repeat_print_wait(capsys: CaptureFixture) -> None:
-    @repeat_every(seconds=0.07, max_repetitions=3, wait_first=0.1)
-    async def repeatedly_print_hello() -> None:
-        print("hello")
 
-    await repeatedly_print_hello()
-    await asyncio.sleep(0.15)
-    out, err = capsys.readouterr()
-    assert out == "hello\n" * 1
-    assert err == ""
+@pytest.fixture(scope="module")
+def wait_first(seconds) -> float:
+    return seconds
 
 
-@pytest.mark.asyncio
-async def test_repeat_unlogged_error(caplog: LogCaptureFixture) -> None:
-    @repeat_every(seconds=0.07, max_repetitions=None)
-    def log_exc() -> NoReturn:
-        raise ValueError("repeat")
+# Tests:
 
-    await log_exc()
-    await asyncio.sleep(0.1)
-    record_tuples = [x for x in caplog.record_tuples if x[0] == __name__]
-    print(caplog.record_tuples)
-    assert len(record_tuples) == 0
 
+class TestRepeatEveryWithSynchronousFunction:
+    def setup_method(self):
+        self.counter = 0

```

Comment:
this setup_method repreats itself- you can create an abstract test class to avoid code duplication

Reply:
It feels a little weird to define a base class for that, especially if it holds not test cases, but I'll trust you with this one.